File: nuido_flow_ai/flows/ai/list_memory_node.py
# THIS FILE IS A PART OF PUBLIC REPOSITORY https://github.com/yonitjio/exploring-odoo
#
# This software is released under the MIT License.
# https://opensource.org/licenses/MIT
#
# THIS SOFTWARE IS EXPERIMENTAL AND FOR EDUCATIONAL PURPOSE ONLY.
# DO NOT USE IT IN PRODUCTION.
from odoo.addons.nuido_flow.flows.core.base_node import BaseNode

from autogen_core.memory import ListMemory, MemoryContent, MemoryMimeType
from .utils import run_async_function

class ListMemoryNode(BaseNode):
    async def _build_memory(self):
        list_memory = ListMemory()

        memories = self.definition["memories"]

        for memory in memories:
            await list_memory.add(MemoryContent(content=memory, mime_type=MemoryMimeType.TEXT))

        return list_memory

    def _process(self, params):
        res = run_async_function(self._build_memory)

        return {
            "memory": res
        }

File: nuido_flow_ai/static/src/models/ai/list_memory_node.js
/*!
// THIS FILE IS A PART OF PUBLIC REPOSITORY https://github.com/yonitjio/exploring-odoo
//
// This software is released under the MIT License.
// https://opensource.org/licenses/MIT
//
// THIS SOFTWARE IS EXPERIMENTAL AND FOR EDUCATIONAL PURPOSE ONLY.
// DO NOT USE IT IN PRODUCTION.
*/
import { NodeModel } from "@nuido/models/node";
import { AiMemoryPort } from "@nuido_flow_ai/components/ports/ai_memory_port";
export class ListMemoryNodeModel extends NodeModel {
    memories;
    setup() {
        const auxOutId = "aux-out-" + this.id + "-list-memory";
        this.addAuxOutPort(auxOutId, AiMemoryPort.name, Number.MAX_SAFE_INTEGER, {
            role: "ai-memory"
        });
        this.memories = [];
    }
    addMemory(value) {
        this.memories.push(value);
    }
    removeMemory(idx) {
        this.memories.splice(idx, 1);
    }
    getData() {
        const res = super.getData();
        return Object.assign(res, {
            memories: this.memories
        });
    }
}
